Update linter docs for dbt Cloud CLI (#5753)
fishtown-analytics/docs.getdbt.com: website/docs/docs/cloud/configure-cloud-cli.md

- `dbt run --help`: Lists the flags available for the `run` command
:::
 
### Lint SQL files 

From the dbt Cloud CLI, you can invoke [SQLFluff](https://sqlfluff.com/) which is a modular and configurable SQL linter that warns you of complex functions, syntax, formatting, and compilation errors. The available SQLFluff commands are: `fix`, `format`, and `lint`. To show detailed information about these commands and the available flags, run the `dbt sqlfluff -h` command. 

> The `lint` command in dbt supports many of the same flags that you can pass to SQLFluff, except for those that might pose a security risk. To lint SQL files, run the command as follows:  

Reviewer: Perhaps, "Running the `lint` command from the dbt Cloud CLI supports many of the same flags that you can pass to SQLFluff." to clarify dbt vs. the cloud cli.

Reviewer 2: +1, might be worth adding that the `fix` and `format` commands, in addition to `lint`, support many of the same flags:
```
Running the supported SQLFluff commands from the dbt Cloud CLI supports many of the same flags that you can pass to SQLFluff, except those that might pose a security risk.

To lint SQL files, run the command as follows:
```

```shell
dbt sqlfluff lint [PATHS]... [flags]
```

> When no path is set, dbt lints all SQL files in the project. To lint a specific SQL file or a directory of SQL files, set `PATHS` to the path of the SQL file or the directory path, respectively. 

Reviewer: "When no path is set, dbt lints all SQL files in the current project. To lint a specific SQL file or a directory, set `PATHS` to the path of the SQL file(s) or directory of files you'd like to lint. You can pass multiple `PATHS` to lint multiple files or directories."

      --rules string                  Narrow the search to only specific rules. For example specifying –-rules LT01 will only search for rule LT01 (Unnecessary trailing whitespace). Multiple rules can be specified with commas e.g. –-rules LT01,LT02 will specify only looking for violations of rule LT01 and rule LT02.
  -t, --templater SqlfluffTemplater   The templater to use, one of "raw", "jinja", "python", "placeholder", "dbt". (default jinja)
  -v, --verbose count                 Verbosity, how detailed should the output be. This is stackable, so -vv is more verbose than -v. For the most verbose option try -vvvv or -vvvvv.

Global Flags:
      --log-format LogFormat   The log format, either json or plain. (default plain)
      --log-level LogLevel     The log level, one of debug, info, warning, error or fatal. (default info)
      --no-color               Disables colorization of the output.
  -q, --quiet                  Suppress all non-error logging to stdout.

> ```

Reviewer 2: This is nice to have here, though if changes are made to the CLI help text this text will also need to be updated or be out of date.

I'd vote to remove this section and only keep the `To show a list of all the dbt supported flags, run the `dbt sqlfluff lint -h` command.` line. Thoughts?

Author: thanks for flagging! i'm pro to this idea! 

i was torn between those two ideas (with/without usage info) but decided to go with being more verbose in this draft to see it all rendered

#### Considerations
Keep the following points in mind when using SQLFluff with dbt Cloud:

- When you run `dbt sqlfluff`, it picks up changes to your local .sqlfluff config.

> - To use SQLFluff in continuous integration/continuous development, you need to have a `dbt_cloud.yml` file in your project and run commands from a valid dbt project.

Reviewer: "Sqlfluff commands run via the cloud cli currently return an exit code of 0 if the invocation successfully ran - rather than the SQLFluff behavior, where a linting violation returns a non-zero exit code. We plan on addressing this in a later release."

Reviewer 2: Agreed, this is a good call out

- `dbt run --help`: Lists the flags available for the `run` command
:::
 
### Lint SQL files 

> From the dbt Cloud CLI, you can invoke [SQLFluff](https://sqlfluff.com/) which is a modular and configurable SQL linter that warns you of complex functions, syntax, formatting, and compilation errors. Many of the same flags that you can pass to SQLFluff are available from the dbt Cloud CLI, except those that might pose a security risk.

Reviewer 3: I don't think we should include that line about a security risk. That's could open us up to potential panicked calls or confusion. The sentence also reads a little weird, I know this isn't the intent, but the quick reader might look at that as us saying that using SQLFluff flags with dbt Cloud CLI is a security risk and other readers might seem like we're calling using SQLFluff a security risk. We could enhance the sentence and give specifics of what might be a risk, but I suggest removing the "security risk" portion entirely.
```suggestion
From the dbt Cloud CLI, you can invoke [SQLFluff](https://sqlfluff.com/) which is a modular and configurable SQL linter that warns you of complex functions, syntax, formatting, and compilation errors. Many of the same flags that you can pass to SQLFluff are available from the dbt Cloud CLI.
```

Author: that's a fair point!

The available SQLFluff commands are: 

- `lint` &mdash; Lint SQL files by passing a list of files or from standard input (stdin).
- `fix` &mdash; Fix SQL files.
- `format` &mdash; Autoformat SQL files.

> To lint SQL files, run the command as follows:  

Reviewer 3: ```suggestion
To lint SQL files run the following command:  
```

```shell
dbt sqlfluff lint [PATHS]... [flags]
```

> When no path is set, dbt lints all SQL files in the current project. To lint a specific SQL file or a directory, set `PATHS` to the path of the SQL file(s) or directory of files. To lint multiple files or directories, pass multiple `PATHS` flags.  

Reviewer 3: ```suggestion
When no path is set, SQLFluff lints all SQL files in the current dbt project. To lint a specific SQL file or a directory, set `PATHS` to the path of the SQL file(s) or directory of files. To lint multiple files or directories, pass multiple `PATHS` flags.  
```

> To show detailed information on all the dbt supported commands and flags, run the `dbt sqlfluff -h` command. 

Reviewer 3: Should we keep this consistent with the example above and put it in a ```shell ``` box?

Author: imo i think this is fine. and there's prior art for it used in a callout on this page. the other line is a command usage vs an example per se

#### Considerations

When running `dbt sqlfluff` from the dbt Cloud CLI, the following are important behaviors to consider:

> - dbt will read the `.sqlfluff` file if it exists for any custom configurations you might have.

Reviewer 3: ```suggestion
- dbt reads the `.sqlfluff` file, if it exists, for any custom configurations you might have.
```

- For continuous integration/continuous development (CI/CD) workflows, your project must have a `dbt_cloud.yml` file and you have successfully run commands from within this dbt project.

> - An SQLFluff command will return an exit code of 0 if it ran without any file violations. This dbt behavior differs from SQLFluff behavior, where a linting violation returns a non-zero exit code. dbt Labs plans on addressing this in a later release.

Reviewer 2: nit
```suggestion
- A SQLFluff command will return an exit code of 0 if it ran without any file violations. This dbt behavior differs from SQLFluff behavior, where a linting violation returns a non-zero exit code. dbt Labs plans on addressing this in a later release.
```

Author: i've been taught to use "Es cue el" vs "sequel" for SQL so "an" is correct

Reviewer 2: I suppose this is a matter of taste. "An" reads a little strange to me and it looks like sqlfluff themselves are using "a sqlfluff" when writing similar sentences - see first bullet under highlights: https://github.com/sqlfluff/sqlfluff/blob/main/CHANGELOG.md#highlights-45

I'll defer to your best judgement though, feel free to proceed as you see fit 😃

Author: yeah, that's true, it's a pref thing. 

you know, i'll go ahead and add to our team meeting agenda this week. whatever we gavel on, i can update our style guide so we can all be consistent

Reviewer 2: sounds good!
